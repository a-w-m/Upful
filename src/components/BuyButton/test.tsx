import React from 'react'
import {render} from '@testing-library/react'
import BuyButton from "."

describe("BuyButton Component", ()=>{
    test("matches snapshot", ()=>{
        const {container} = render( <BuyButton
            className ='snipcart-add-item'
            data-item-id= ""
            data-item-price= ""
            data-item-name= ""
            data-item-description=""
            data-item-image=""
            data-item-url=""
            data-item-custom1-name=""
            data-item-custom1-options=""
            data-item-custom1-value="">
       </BuyButton>)
        expect(container.firstChild).toMatchSnapshot()

    })
})