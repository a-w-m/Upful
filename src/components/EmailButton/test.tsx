import React from "react"
import { render } from "@testing-library/react"
import EmailButton from "."

const props ={
    title: "test-email"
}

describe("EmailButton component", () => {
  test("matches snapshot", () => {
    const { container } = render(<EmailButton {...props}></EmailButton>)
    expect(container.firstChild).toMatchSnapshot()
  })


  test("component displays correct text", ()=>{
    const { queryByText } = render(<EmailButton {...props}></EmailButton>)
    expect(queryByText("Email us about this product")).toBeTruthy();

  })

  test("component has correct href", ()=>{
    const { getByRole } = render(<EmailButton {...props}></EmailButton>)

    // Click button
    expect(getByRole('link')).toHaveAttribute("href", `mailto:howdy@upful.com?Subject=${props.title}`)


  })

 
})
