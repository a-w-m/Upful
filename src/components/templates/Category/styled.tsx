import styled from 'styled-components'

export const CategoryHeading = styled.h1`
font-size: var(--heading-font-size);
font-family: var(--main-font-size);
padding-bottom: 2rem;
text-align: center;
`

export const Container = styled.div`
display: grid;
grid-auto-rows: auto;
grid-gap: 1rem;
`