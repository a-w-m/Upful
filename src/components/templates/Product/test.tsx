import React from 'react'
import {render} from '@testing-library/react'
import Product from "./index"

const data =   {
    markdownRemark: {
    frontmatter:{
      
      title: "",
      price: "",
      image: "",
      id: "",
      description: "",
      date: "",
      customField: {
        name: "",
        values: [ {name: "", priceChange: "" }]
      }
    }
  }
  }

describe("Item Component", ()=>{
    test("matches snapshot", ()=>{
        const {container} = render(<Product data = {data} pageContext = {{slug: ""}}/>)
        expect(container.firstChild).toMatchSnapshot()

    })
})