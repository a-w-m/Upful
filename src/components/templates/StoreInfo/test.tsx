import React from "react"
import { render, screen } from "@testing-library/react"
import Markdown from "./index"
const mockData = {
  data: {
    file: {
      childMarkdownRemark: {
        html: "<p>TEST</p>",
      },
    },
  },
}

describe("Markdown component", () => {
  test("snapshot", () => {
    const { container } = render(<Markdown {...mockData} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test("render html", () => {
    render(<Markdown {...mockData} />)
    const html = screen.getByText("TEST")
    expect(html).toBeInTheDocument()
  })
})
