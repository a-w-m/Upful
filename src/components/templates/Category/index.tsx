import React from "react"
import Layout from "../../Layout"
import Seo from "../../Seo"
import { graphql } from "gatsby"
import ProductGrid from "../../ProductGrid"
import Pagination from "../../Pagination"
import { CategoryHeading, Container } from "./styled"
import { C } from "../../../interfaces"

const Category: React.FC<C.CategoryPage> = ({ data, pageContext }) => {
  const { category, numPages, currentPage } = pageContext
  return (
    <Layout>
      <Seo />
      <Container>
        <CategoryHeading>{category}</CategoryHeading>
        <ProductGrid edges={data.allFile.edges} />
        {numPages > 1 && (
          <Pagination
            currentPage={currentPage}
            numPages={numPages}
            category={category}
          />
        )}
      </Container>
    </Layout>
  )
}

export const query = graphql`
  query ($skip: Int! = 0, $category: String!, $limit: Int!) {
    allFile(
      filter: {
        sourceInstanceName: { eq: $category }
        internal: { mediaType: { eq: "text/markdown" } }
      }
      limit: $limit
      skip: $skip
    ) {
      edges {
        node {
          childMarkdownRemark {
            ...Thumbnail
            fields {
              slug
            }
          }
        }
      }
    }
  }
`

export default Category
