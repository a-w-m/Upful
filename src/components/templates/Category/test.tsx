import React from "react"
//import custom render
import { render, screen } from "src/utils/test/test-utils"
import Category from "./index"
import { getProductGridData } from "src/utils/test/data"

const mockData = {
  allFile: { ...getProductGridData() },
}

const mockPageContext = {
  currentPage: 1,
  category: "Test",
  limit: 4,
  skip: 0,
  numPages: 1,
}

describe("Category", () => {
  it("matches snapshot", () => {
    const { container } = render(
      <Category data={mockData} pageContext={mockPageContext}></Category>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it("should display category heading", () => {
    render(<Category data={mockData} pageContext={mockPageContext}></Category>)
    const heading = screen.queryByRole("heading", { level: 1 })
    expect(heading).toBeInTheDocument()
  })

  it("should not display pagination if there is only one page of products", () => {
    render(<Category data={mockData} pageContext={mockPageContext}></Category>)
    const pagination = screen.queryByText("/Page/")
    expect(pagination).toBeNull()
  })

  it("should display pagination if there is more than one page of products", () => {
    render(
      <Category
        data={mockData}
        pageContext={{ ...mockPageContext, numPages: 2 }}
      ></Category>
    )
    const pagination = screen.getByText("Page 1 of 2")
    expect(pagination).toBeInTheDocument
  })

  it("should display all products in the category", () => {
    render(<Category data={mockData} pageContext={mockPageContext}></Category>)
    const products = screen.getAllByRole("article")
    expect(products).toHaveLength(mockData.allFile.edges.length)
  })
})
