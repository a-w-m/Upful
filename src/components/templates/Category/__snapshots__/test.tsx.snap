// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category matches snapshot 1`] = `
<div
  class="sc-dIouRR DzqZm"
>
  <header
    class="sc-jSMfEi fWJutG"
  >
    <div
      class="sc-gKXOVf bRgbNH"
    >
      <div
        class="sc-bczRLJ iKPXgD"
      >
        <button
          aria-controls="menu"
          aria-expanded="false"
          aria-label="toggle navigation"
          class="sc-gsnTZi jUeZLU"
          type="button"
        >
          <span />
          <span />
          <span />
        </button>
      </div>
      <nav
        class="sc-dkzDqf ckAmYP"
        id="menu"
      >
        <a
          class="sc-eCYdqJ kRVJaY"
          href="/"
        >
          <div
            class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
            data-gatsby-image-wrapper=""
            style="position: relative; overflow: hidden; display: inline-block; vertical-align: top; border-radius: 100%;"
          >
            <div
              style="max-width: 250px; display: block;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;charset=utf-8,%3Csvg height='250' width='250' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                style="max-width: 100%; display: block; position: static;"
              />
            </div>
            <div
              aria-hidden="true"
              data-placeholder-image=""
              style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            />
            <picture>
              <source
                data-srcset="/static/55955cd5db9ac5b9496eb4fb43c87585/74c72/upful-gold-frame-logo.webp 63w,/static/55955cd5db9ac5b9496eb4fb43c87585/d66e1/upful-gold-frame-logo.webp 125w,/static/55955cd5db9ac5b9496eb4fb43c87585/e7160/upful-gold-frame-logo.webp 250w,/static/55955cd5db9ac5b9496eb4fb43c87585/5f169/upful-gold-frame-logo.webp 500w"
                sizes="(min-width: 250px) 250px, 100vw"
                type="image/webp"
              />
              <img
                alt="logo"
                data-gatsby-image-ssr=""
                data-main-image=""
                data-src="/static/55955cd5db9ac5b9496eb4fb43c87585/de391/upful-gold-frame-logo.png"
                data-srcset="/static/55955cd5db9ac5b9496eb4fb43c87585/6a639/upful-gold-frame-logo.png 63w,/static/55955cd5db9ac5b9496eb4fb43c87585/2fd20/upful-gold-frame-logo.png 125w,/static/55955cd5db9ac5b9496eb4fb43c87585/de391/upful-gold-frame-logo.png 250w,/static/55955cd5db9ac5b9496eb4fb43c87585/82c11/upful-gold-frame-logo.png 500w"
                decoding="async"
                loading="lazy"
                sizes="(min-width: 250px) 250px, 100vw"
                style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
              />
            </picture>
            <noscript />
            <script
              type="module"
            >
              const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
            </script>
          </div>
        </a>
        <a
          class="sc-hKMtZM cdOabJ"
          href="/women/"
        >
          WOMEN
        </a>
        <a
          class="sc-hKMtZM cdOabJ"
          href="/kids/"
        >
          KIDS
        </a>
      </nav>
    </div>
    <a
      class="sc-eCYdqJ kRVJaY"
      href="/"
    >
      <div
        class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
        data-gatsby-image-wrapper=""
        style="position: relative; overflow: hidden; display: inline-block; vertical-align: top; border-radius: 100%;"
      >
        <div
          style="max-width: 250px; display: block;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;charset=utf-8,%3Csvg height='250' width='250' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
            style="max-width: 100%; display: block; position: static;"
          />
        </div>
        <div
          aria-hidden="true"
          data-placeholder-image=""
          style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
        />
        <picture>
          <source
            data-srcset="/static/55955cd5db9ac5b9496eb4fb43c87585/74c72/upful-gold-frame-logo.webp 63w,/static/55955cd5db9ac5b9496eb4fb43c87585/d66e1/upful-gold-frame-logo.webp 125w,/static/55955cd5db9ac5b9496eb4fb43c87585/e7160/upful-gold-frame-logo.webp 250w,/static/55955cd5db9ac5b9496eb4fb43c87585/5f169/upful-gold-frame-logo.webp 500w"
            sizes="(min-width: 250px) 250px, 100vw"
            type="image/webp"
          />
          <img
            alt="logo"
            data-gatsby-image-ssr=""
            data-main-image=""
            data-src="/static/55955cd5db9ac5b9496eb4fb43c87585/de391/upful-gold-frame-logo.png"
            data-srcset="/static/55955cd5db9ac5b9496eb4fb43c87585/6a639/upful-gold-frame-logo.png 63w,/static/55955cd5db9ac5b9496eb4fb43c87585/2fd20/upful-gold-frame-logo.png 125w,/static/55955cd5db9ac5b9496eb4fb43c87585/de391/upful-gold-frame-logo.png 250w,/static/55955cd5db9ac5b9496eb4fb43c87585/82c11/upful-gold-frame-logo.png 500w"
            decoding="async"
            loading="lazy"
            sizes="(min-width: 250px) 250px, 100vw"
            style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
          />
        </picture>
        <noscript />
        <script
          type="module"
        >
          const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
        </script>
      </div>
    </a>
    <div
      class="sc-kDDrLX fsOTBx"
    >
      <button
        aria-label="snipcart-checkout"
        class="sc-iqcoie ingRrx snipcart-checkout"
        type="button"
      >
        <svg
          class="snipcart-cart-header__icon   snipcart__icon"
          fill="none"
          viewBox="0 0 64 64"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M51.714 20.47L55 60H9l3.286-39.53h9.857v-6.588C22.143 8.424 26.556 4 32 4c5.444 0 9.857 4.424 9.857 9.882v6.589h9.857zM25.43 13.883v16.47h-3.286v-6.587h-6.834l-2.737 32.94h38.856l-2.737-32.94h-6.834v6.588h-3.286v-16.47c0-3.634-2.947-6.589-6.571-6.589-3.624 0-6.571 2.955-6.571 6.588zm3.285 9.883V20.47h6.572v3.294h-6.572z"
            fill="#81550d"
            fill-rule="evenodd"
          />
        </svg>
      </button>
      <span
        class="sc-crXcEl iMYtNW"
      >
        0
      </span>
    </div>
  </header>
  <div
    class="sc-hHLeRK bMSYlC"
  >
    <main
      class="sc-dmRaPn"
    >
      <div
        class="sc-ezWOiH kRVgym"
      >
        <h1
          class="sc-gicCDI cYQkMt"
        >
          Test
        </h1>
        <div
          class="sc-llJcti InoPk"
        >
          <article
            class="sc-kgflAQ cjynRD"
          >
            <a
              href="/easy-sweatpant-in-natural/"
            >
              <div
                class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
                data-gatsby-image-wrapper=""
                style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
              >
                <div
                  style="max-width: 1030px; display: block;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;charset=utf-8,%3Csvg height='1030' width='1030' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                    style="max-width: 100%; display: block; position: static;"
                  />
                </div>
                <div
                  aria-hidden="true"
                  data-placeholder-image=""
                  style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
                />
                <img
                  alt="thumbnail"
                  data-gatsby-image-ssr=""
                  data-main-image=""
                  data-src="/static/1a4732d2748806c6a0c45dcd0ac92ca4/a640e/image-1.webp"
                  data-srcset="/static/1a4732d2748806c6a0c45dcd0ac92ca4/c7cc5/image-1.webp 258w,/static/1a4732d2748806c6a0c45dcd0ac92ca4/f26c5/image-1.webp 515w,/static/1a4732d2748806c6a0c45dcd0ac92ca4/a640e/image-1.webp 1030w"
                  decoding="async"
                  loading="lazy"
                  sizes="(min-width: 1030px) 1030px, 100vw"
                  style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
                />
                <noscript />
                <script
                  type="module"
                >
                  const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
                </script>
              </div>
            </a>
            <div
              class="sc-cxabCf ewvQAt"
            >
              <h3
                class="sc-fLlhyt eAzYIh"
              >
                Easy Sweatpant in Natural
              </h3>
              <div
                class="sc-bBrHrO iqfpEJ"
              >
                158.00
              </div>
              <span
                class="sc-ivTmOn"
              >
                Sold Out
              </span>
            </div>
          </article>
          <article
            class="sc-kgflAQ cjynRD"
          >
            <a
              href="/haori-coat-in-washed-denim/"
            >
              <div
                class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
                data-gatsby-image-wrapper=""
                style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
              >
                <div
                  style="max-width: 995px; display: block;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;charset=utf-8,%3Csvg height='995' width='995' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                    style="max-width: 100%; display: block; position: static;"
                  />
                </div>
                <div
                  aria-hidden="true"
                  data-placeholder-image=""
                  style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
                />
                <img
                  alt="thumbnail"
                  data-gatsby-image-ssr=""
                  data-main-image=""
                  data-src="/static/5e566884fdc0f35caf38e43e3d3b3faa/87640/image-1.webp"
                  data-srcset="/static/5e566884fdc0f35caf38e43e3d3b3faa/dc55b/image-1.webp 249w,/static/5e566884fdc0f35caf38e43e3d3b3faa/525ed/image-1.webp 498w,/static/5e566884fdc0f35caf38e43e3d3b3faa/87640/image-1.webp 995w"
                  decoding="async"
                  loading="lazy"
                  sizes="(min-width: 995px) 995px, 100vw"
                  style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
                />
                <noscript />
                <script
                  type="module"
                >
                  const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
                </script>
              </div>
            </a>
            <div
              class="sc-cxabCf ewvQAt"
            >
              <h3
                class="sc-fLlhyt eAzYIh"
              >
                Haori Coat in Washed Denim
              </h3>
              <div
                class="sc-bBrHrO iTHiwH"
              >
                420.00
              </div>
            </div>
          </article>
          <article
            class="sc-kgflAQ cjynRD"
          >
            <a
              href="/mardi-dress-in-majorelle/"
            >
              <div
                class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
                data-gatsby-image-wrapper=""
                style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
              >
                <div
                  style="max-width: 1989px; display: block;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;charset=utf-8,%3Csvg height='1989' width='1989' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                    style="max-width: 100%; display: block; position: static;"
                  />
                </div>
                <div
                  aria-hidden="true"
                  data-placeholder-image=""
                  style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
                />
                <img
                  alt="thumbnail"
                  data-gatsby-image-ssr=""
                  data-main-image=""
                  data-src="/static/9bb4bfc8c3b86199d4c6f2ce7bd2fe68/e565f/image-1.webp"
                  data-srcset="/static/9bb4bfc8c3b86199d4c6f2ce7bd2fe68/ce874/image-1.webp 497w,/static/9bb4bfc8c3b86199d4c6f2ce7bd2fe68/87640/image-1.webp 995w,/static/9bb4bfc8c3b86199d4c6f2ce7bd2fe68/e565f/image-1.webp 1989w"
                  decoding="async"
                  loading="lazy"
                  sizes="(min-width: 1989px) 1989px, 100vw"
                  style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
                />
                <noscript />
                <script
                  type="module"
                >
                  const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
                </script>
              </div>
            </a>
            <div
              class="sc-cxabCf ewvQAt"
            >
              <h3
                class="sc-fLlhyt eAzYIh"
              >
                Mardi Dress in Majorelle
              </h3>
              <div
                class="sc-bBrHrO iqfpEJ"
              >
                330.00
              </div>
            </div>
          </article>
          <article
            class="sc-kgflAQ cjynRD"
          >
            <a
              href="/mona-marble-dress/"
            >
              <div
                class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
                data-gatsby-image-wrapper=""
                style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
              >
                <div
                  style="max-width: 832px; display: block;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;charset=utf-8,%3Csvg height='832' width='832' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                    style="max-width: 100%; display: block; position: static;"
                  />
                </div>
                <div
                  aria-hidden="true"
                  data-placeholder-image=""
                  style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
                />
                <img
                  alt="thumbnail"
                  data-gatsby-image-ssr=""
                  data-main-image=""
                  data-src="/static/3ab4e7845d0eddea1b17acedc963dd31/8f66e/image-1.webp"
                  data-srcset="/static/3ab4e7845d0eddea1b17acedc963dd31/c95dc/image-1.webp 208w,/static/3ab4e7845d0eddea1b17acedc963dd31/3f9a3/image-1.webp 416w,/static/3ab4e7845d0eddea1b17acedc963dd31/8f66e/image-1.webp 832w"
                  decoding="async"
                  loading="lazy"
                  sizes="(min-width: 832px) 832px, 100vw"
                  style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
                />
                <noscript />
                <script
                  type="module"
                >
                  const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
                </script>
              </div>
            </a>
            <div
              class="sc-cxabCf ewvQAt"
            >
              <h3
                class="sc-fLlhyt eAzYIh"
              >
                Mona Marble Dress
              </h3>
              <div
                class="sc-bBrHrO iqfpEJ"
              >
                280.00
              </div>
            </div>
          </article>
          <article
            class="sc-kgflAQ cjynRD"
          >
            <a
              href="/the-field-jumpsuit/"
            >
              <div
                class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
                data-gatsby-image-wrapper=""
                style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
              >
                <div
                  style="max-width: 1024px; display: block;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;charset=utf-8,%3Csvg height='1024' width='1024' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                    style="max-width: 100%; display: block; position: static;"
                  />
                </div>
                <div
                  aria-hidden="true"
                  data-placeholder-image=""
                  style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
                />
                <img
                  alt="thumbnail"
                  data-gatsby-image-ssr=""
                  data-main-image=""
                  data-src="/static/01ab4798217057a46bbd2c052b23d4bd/67ded/image-1.webp"
                  data-srcset="/static/01ab4798217057a46bbd2c052b23d4bd/22bfc/image-1.webp 256w,/static/01ab4798217057a46bbd2c052b23d4bd/d689f/image-1.webp 512w,/static/01ab4798217057a46bbd2c052b23d4bd/67ded/image-1.webp 1024w"
                  decoding="async"
                  loading="lazy"
                  sizes="(min-width: 1024px) 1024px, 100vw"
                  style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
                />
                <noscript />
                <script
                  type="module"
                >
                  const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
                </script>
              </div>
            </a>
            <div
              class="sc-cxabCf ewvQAt"
            >
              <h3
                class="sc-fLlhyt eAzYIh"
              >
                The Field Jumpsuit
              </h3>
              <div
                class="sc-bBrHrO iqfpEJ"
              >
                145.00
              </div>
            </div>
          </article>
          <article
            class="sc-kgflAQ cjynRD"
          >
            <a
              href="/virginia-dress/"
            >
              <div
                class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
                data-gatsby-image-wrapper=""
                style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
              >
                <div
                  style="max-width: 2004px; display: block;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;charset=utf-8,%3Csvg height='2004' width='2004' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                    style="max-width: 100%; display: block; position: static;"
                  />
                </div>
                <div
                  aria-hidden="true"
                  data-placeholder-image=""
                  style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
                />
                <img
                  alt="thumbnail"
                  data-gatsby-image-ssr=""
                  data-main-image=""
                  data-src="/static/9bc7109a20b6a0d06cfdc4e6bb8a3970/c7f64/image-1.webp"
                  data-srcset="/static/9bc7109a20b6a0d06cfdc4e6bb8a3970/d9828/image-1.webp 501w,/static/9bc7109a20b6a0d06cfdc4e6bb8a3970/a5c80/image-1.webp 1002w,/static/9bc7109a20b6a0d06cfdc4e6bb8a3970/c7f64/image-1.webp 2004w"
                  decoding="async"
                  loading="lazy"
                  sizes="(min-width: 2004px) 2004px, 100vw"
                  style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
                />
                <noscript />
                <script
                  type="module"
                >
                  const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
                </script>
              </div>
            </a>
            <div
              class="sc-cxabCf ewvQAt"
            >
              <h3
                class="sc-fLlhyt eAzYIh"
              >
                Virginia Dress
              </h3>
              <div
                class="sc-bBrHrO iqfpEJ"
              >
                440.00
              </div>
            </div>
          </article>
          <article
            class="sc-kgflAQ cjynRD"
          >
            <a
              href="/peace-daisy-tee/"
            >
              <div
                class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
                data-gatsby-image-wrapper=""
                style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
              >
                <div
                  style="max-width: 1000px; display: block;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;charset=utf-8,%3Csvg height='1000' width='1000' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                    style="max-width: 100%; display: block; position: static;"
                  />
                </div>
                <div
                  aria-hidden="true"
                  data-placeholder-image=""
                  style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
                />
                <img
                  alt="thumbnail"
                  data-gatsby-image-ssr=""
                  data-main-image=""
                  data-src="/static/63315f2eafa8f2947de0de1c5d3a32dd/3cd29/image-1.webp"
                  data-srcset="/static/63315f2eafa8f2947de0de1c5d3a32dd/e7160/image-1.webp 250w,/static/63315f2eafa8f2947de0de1c5d3a32dd/5f169/image-1.webp 500w,/static/63315f2eafa8f2947de0de1c5d3a32dd/3cd29/image-1.webp 1000w"
                  decoding="async"
                  loading="lazy"
                  sizes="(min-width: 1000px) 1000px, 100vw"
                  style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
                />
                <noscript />
                <script
                  type="module"
                >
                  const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
                </script>
              </div>
            </a>
            <div
              class="sc-cxabCf ewvQAt"
            >
              <h3
                class="sc-fLlhyt eAzYIh"
              >
                Peace Daisy Tee
              </h3>
              <div
                class="sc-bBrHrO iqfpEJ"
              >
                55.00
              </div>
            </div>
          </article>
        </div>
      </div>
    </main>
  </div>
  <footer
    class="sc-evZas NxEgn"
  >
    <nav
      class="sc-breuTD cHRYoe"
    >
      <a
        class="sc-ksZaOG"
        href="/women/"
      >
        women
      </a>
      <a
        class="sc-ksZaOG"
        href="/kids/"
      >
        kids
      </a>
      <a
        class="sc-ksZaOG"
        href="/about/"
      >
        about
      </a>
      <a
        class="sc-ksZaOG"
        href="/returns/"
      >
        returns
      </a>
    </nav>
    <address
      class="sc-idiyUo gwMrhD"
    >
      <a
        href="mailto:howdy@storefront.com"
      >
        howdy@storefront.com
      </a>
    </address>
    <div
      class="sc-fEOsli iqDUsf"
    >
      © Upful 
      2022
       Made with full Irations
    </div>
  </footer>
</div>
`;
